File: day1/main.py
# part 1
left = []
right = []
with open("input.txt") as f:
    
    lines = f.readlines()
    for line in lines:
        nums = line.split()
        left.append(int(nums[0]))
        right.append(int(nums[1]))
    left.sort()
    right.sort()

    out = 0
    for x, y in zip(left, right):
        out += abs(x - y)
    print(out)

# part 2
l = []
rfreq = {}
with open("input.txt") as f:
    lines = f.readlines()
    for line in lines:
        nums = line.split()
        nums[0] = int(nums[0])
        nums[1] = int(nums[1])
        l.append(nums[0])
        rfreq[nums[1]] = rfreq.get(nums[1], 0) + 1
    out = 0
    for v in l:
        out += v * rfreq.get(v, 0)
    print(out)

File: day1/main.rs
use std::fs::File;
use std::io::BufReader;
use std::io::prelude::*;

fn part1() -> std::io::Result<()> {
    let f = File::open("input.txt")?;
    let reader = BufReader::new(f);
    for line in reader.lines() {
        let line = line?;
        let mut nums = line.split_whitespace();
        let num1: i32 = nums.next().unwrap().parse().unwrap();
        let num2: i32 = nums.next().unwrap().parse().unwrap();
        println!("{} {}", num1, num2);
    }

    Ok(())

}


fn main() {
    let _ = part1();
}